[#5389] feat(logging): add org.apache.logging.log4j:log4j-layout-template-json to support server log in json format

puchengy opened the pull request:
### What changes were proposed in this pull request?

Add org.apache.logging.log4j:log4j-layout-template-json to support server log in json format.

### Why are the changes needed?

There are use cases required server logs to be viewed from logsearch tools such as Kibana. These use cases recommends log format to be json.

Based on https://logging.apache.org/log4j/2.x/manual/json-template-layout.html, `log4j-layout-template-json` is needed to enable json layout for log4j.

Fix: https://github.com/apache/gravitino/issues/5389

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Tested locally with modified lo4j property file.
```
appender.rolling.layout.type = JsonTemplateLayout
appender.rolling.layout.eventTemplate = {"instant":{"$resolver":"timestamp","pattern":{"format":"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'","timeZone":"UTC"}},"level":{"$resolver":"level","field":"name"},"thread":{"$resolver":"thread","field":"name"},"message":{"$resolver":"message","stringified":true},"exception":{"$resolver":"exception","field":"stackTrace","stackTrace":{"stringified":true}}}
```

For better usability, json can be passed in via a individual file using [eventTemplateUri](https://logging.apache.org/log4j/2.x/manual/json-template-layout.html#plugin-attr-eventTemplateUri) instead.

jerryshao commented:
Hi @puchengy thanks a lot for your contribution. Can you please create a related issue about this, also change the PR title and description to map to the related issue? Thanks a lot.

puchengy commented:
@jerryshao for sure, done.

FANNG1 commented:
LGTM, @puchengy , could you provide an json configuration example in the PR description, then the other user could reffer to it?

jerryshao commented:
Yes, we should, @puchengy can you please update the "LICENSE.bin" for your newly added dependency, thanks!

puchengy commented:
@FANNG1 @yuqi1129 @jerryshao Thanks all. Added a json configuration in the test plan. Also added dep to licensen.bin, the name of the library is obtained from https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-layout-template-json. PTAL, thanks
